kernel/src/snapshot/builder.rs

```diff
@@ -0,0 +1,169 @@
+//! Builder for creating [`Snapshot`] instances.
+
+use crate::log_segment::LogSegment;
+use crate::{DeltaResult, Engine, Snapshot, Version};
+
+use url::Url;
+
+/// Builder for creating [`Snapshot`] instances.
+///
+/// # Example
+///
+/// ```no_run
+/// # use delta_kernel::{Snapshot, Engine};
+/// # use url::Url;
+/// # fn example(engine: &dyn Engine) -> delta_kernel::DeltaResult<()> {
+/// let table_root = Url::parse("file:///path/to/table")?;
+///
+/// // Build a snapshot at the latest version
+/// let snapshot = Snapshot::build(table_root.clone()).build_latest(engine)?;
+///
+/// // Build a snapshot at a specific version (time travel)
+/// let snapshot = Snapshot::build(table_root).build_at(5, engine)?;
+/// # Ok(())
+/// # }
+/// ```
+pub struct SnapshotBuilder {
+    table_root: Url,
+}
+
+impl SnapshotBuilder {
+    pub(crate) fn new(table_root: Url) -> Self {
+        Self { table_root }
+    }
+
+    /// Create a new [`Snapshot`] instance for a specific version of the table.
+    ///
+    /// # Parameters
+    ///
+    /// - `version`: target version of the [`Snapshot`].
+    /// - `engine`: implementation of [`Engine`] apis.
+    pub fn build_at(self, version: Version, engine: &dyn Engine) -> DeltaResult<Snapshot> {
+        self.build(Some(version), engine)
+    }
+
+    /// Create a new [`Snapshot`] instance for the latest version of the table.
+    ///
+    /// # Parameters
+    ///
+    /// - `engine`: Implementation of [`Engine`] apis.
+    pub fn build_latest(self, engine: &dyn Engine) -> DeltaResult<Snapshot> {
+        self.build(None, engine)
+    }
```

Comment:
What are the options we expect to configure in the future for SnapshotBuilder? Presently these two methods can just be part of Snapshot, but I expect there's stuff we don't see in this PR :)

Reply:
Ah! mb
>  since we will in the future allow optimizations like passing a log_tail to the builder or Protocol/Metadata/Version that the catalog has kept around.